test/e2e_node/garbage_collector_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("GarbageCollect", func() {
+	var cl *client.Client
+	var dockerClient *docker.Client
+	BeforeEach(func() {
+		cl = client.NewOrDie(&client.Config{Host: *apiServerAddress})
+		var err error
+		dockerClient, err = docker.NewClientFromEnv()
+		Expect(err).To(BeNil(), fmt.Sprintf("Error connecting to docker %v", err))
+	})
+
+	It("Should garbage collect deleted pods", func() {
+		Skip("Requires docker permissions") // FIXME
+
+		const (
+			// The acceptable delta when counting containers.
+			epsilon = 5
+			// The number of pods to create & delete.
+			numPods = 90
+		)
+
+		containers, err := dockerClient.ListContainers(docker.ListContainersOptions{All: true})
+		Expect(err).To(BeNil(), fmt.Sprintf("Error listing containers %v", err))
+		initialContainerCount := len(containers)
+
+		// Start pods.
+		podNames := make([]string, numPods)
+		podContainers := []api.Container{getPauseContainer()}
+		for i := 0; i < numPods; i++ {
```

Comment:
I think we can reuse https://github.com/kubernetes/kubernetes/blob/5b415a7aede5338595b06e4f4625736b18c185e3/test/e2e/util.go#L1940